hack/lib/version.sh

```diff
@@ -36,6 +36,19 @@ kube::version::get_version_vars() {
     return
   fi
 
+  # If the kubernetes source was exported through git archive, then
+  # we likely don't have a git tree, but these magic values may be filled in.
+  if [[ '$Format:%%$' == "%" ]]; then
+    KUBE_GIT_COMMIT='$Format:%H$'
+    KUBE_GIT_TREE_STATE="not a git tree"
```

Comment:
wonder if this should be "git archive" instead of "not a git tree"?

Reply:
+1

Reply:
[Precedence](https://github.com/kubernetes/kubernetes/blob/master/pkg/version/base.go#L56), though we could change it.  Maybe in both locations?

Reply:
did we want to change this?

Reply:
To git archive or empty?

Reply:
`git archive`

Reply:
done.